src/GraphCoarsen.cpp

```diff
@@ -301,19 +391,27 @@ void GraphCoarsen::ComputeEdgeTargets(const MixedMatrix& mgl,
         const auto& face_M = shared_M[face];
         const auto& face_D = shared_D[face];
         const auto& face_sigma = shared_sigma[face];
+        const auto& face_constant = shared_constant[face];
 
         linalgcpp::HStack(face_sigma, collected_sigma);
 
         bool shared = face_shared.RowSize(face) > 0;
 
         // TODO(gelever1): resolve this copy.  (Types must match so face_? gets copied and promoted
         // to rvalue).
-        const auto& M_local = shared ? Combine(face_M, num_face_edges) : face_M[0];
-        const auto& D_local = shared ? Combine(face_D, num_face_edges) : face_D[0];
+        const auto& M_local = shared ? CombineM(face_M, num_face_edges) : face_M[0];
+        const auto& D_local = shared ? CombineD(face_D, num_face_edges) : face_D[0];
+        const auto& constant_local = shared ? CombineConstant(face_constant) : face_constant[0];
```

Comment:
I think you can make `one_neg_one` directly, bypassing the step of making `constant_local`.